plugins/filter/src/filterlist.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * This file is a part of the Voice Call Manager project
+ *
+ * Copyright (C) 2024  Damien Caliste <dcaliste@free.fr>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "filterlist.h"
+
+#include <MGConfItem>
+
+class FilterList::Private
+{
+public:
+    Private(const QString &key)
+        : m_conf(key)
+    {
+    }
+
+    MGConfItem m_conf;
+};
+
+FilterList::FilterList(const QString &key, QObject *parent)
+    : QObject(parent)
+    , d(new Private(key))
+{
+    connect(&d->m_conf, &MGConfItem::valueChanged,
+            this, &FilterList::changed);
+}
+
+FilterList::~FilterList()
+{
+}
+
+QString FilterList::key() const
+{
+    return d->m_conf.key();
+}
+
+QStringList FilterList::list() const
+{
+    return d->m_conf.value().toStringList();
+}
+
+void FilterList::set(const QStringList &list)
+{
+    d->m_conf.set(list);
+}
+
+bool FilterList::match(const QString &number)
+{
+    return list().contains(number);
```

Comment:
It's a start, but ideally the phone numbers should be matched a bit more smart to make sure numbers with and without the plus prefixed country code get a match.

Maybe qtcontacts-sqlite / qtcontacts-extensions_impl.h might help here, libcommhistory also having some wrapper. Don't remember immediately how this should all go.

Reply:
Indeed, it's just a placeholder for the moment, working more on the changes required in the stack than the implementation of the blocking itself. I'm sure I'll need something like a regexp anyway since French spammers are using ranges of phone numbers.

I was thinking about libphonenumber to do the actual job of matching numbers. But I didn't look at all at their API for the moment.

Reply:
The regexp is probably too techy to expose on the UI so if wanted that could be maybe a different config key or even a separate plugin.

Reply:
I was going to offer some slightly too generic guesses for what is needed (exact and prefix matching only - after normalization). Heck, even exact only might be fine just to get it out the door.
Then i started looking at the libphonenumber... 
I found found some red herrings (e.g. matching number inside text).
Then i found something decently promising (RegexBasedMatcher::Match with allow_prefix_match), only to find that it's not part of the installed headers.
Nor are the building blocks to remake it despite it being trivial. :(

Now i just want to go cry in a corner and feel ashamed to prefer C++ over C.
What a horrible mess they have made. So much noise for basically no functionality beyond the parsing.

...so looping back around. Exact match is done with:
```c++
PhoneNumberUtil* util = PhoneNumberUtil::GetInstance();
util->IsNumberMatchWithTwoStrings(number1, number2)==PhoneNumberUtil::EXACT_MATCH
```
It does not have a lot of wiggle-room/help in the parsing from what i can understand - so what is the point?
Are we just opening up to inability to compare (and thus block) number formats that the current version of libphonenumber considers valid?
(see: https://forum.sailfishos.org/t/cant-send-sms-when-number-too-long/18544)

Reply:
> The regexp is probably too techy to expose on the UI 

Yes sure. I was not clear in my reply. I don't mean to expose regexp in the UI. That's the beauty of discoupling engine and UI. The filtering plugin can handle regexp while the UI is exposing something simple like a context menu in phone log saying "block this number" for instance. Or one can even provide a dedicated "advanced" application that can expose simple matching like suggested by @attah with prefix matching...

The current plugin is based on DConf values. It's easy to implement something that can handle:
```
ignored_numbers = ["+33123456789", "regexp:^{\+33,0}1234567[0-9][0-9]"]
whitelist_numbers = ["+33123456700", "0123456700"]
```
A simple UI integration like the one mentioned before can add or remove numbers like "+33123456789", while an advance app or a techy user can add more complex patterns.

@attah, thanks for your look into libphonenumber. It doesn't sound promising for our application. Seems more oriented for parsing or validating numbers. Doesn't matter, using Qt regexp internally can be largely sufficient after all.